### PR TITLE
Fix bug in ravel_multi_index for big indices (Issue #7546)

### PR DESCRIPTION
**Pull request by skwbc**
I fixed bug on Issue #7546 .
##### 1st commit:

As @jaimefrio said, I just changed type of variable `s` from `int` to `npy_intp`.
##### 2nd commit (not confident):

I added checking for array size overflow using npy_mul_with_overflow_intp, similar to PR #7463 .
It's my understanding that in PR #7463 @seberg checked wheather `arr.size * arr.dtype.itemsize` is greater than `np.iinfo(np.intp).max`.
However, in ravel_multi_index we don't know the type of the array defined by `dims` argument.
So, I just checked whether `arr.size` is greater than `np.iinfo(np.intp).max`. Is this fix okay?

### CODE REVIEW

File: numpy/core/src/multiarray/compiled_base.c [hunk 1]
```diff
@@ -1005,14 +1006,24 @@ arr_ravel_multi_index(PyObject *self, PyObject *args, PyObject *kwds)
             s = 1;
             for (i = dimensions.len-1; i >= 0; --i) {
                 ravel_strides[i] = s;
-                s *= dimensions.ptr[i];
+                if (npy_mul_with_overflow_intp(&s, s, dimensions.ptr[i])) {
+                    PyErr_SetString(PyExc_ValueError,
+                    "invalid dims: array size defined by dims is larger "
```
Comment (seberg) [inline numpy/core/src/multiarray/compiled_base.c]: Extra level of indentation would be nice.

**Review by seberg — APPROVED** [review]
Seems good to me, could fix that indentation (best squashed into the other commites).
We don't have to make sure that the index is valid for a specific array here after all, we just have to make sure we don't get a wrong result.

File: numpy/core/src/multiarray/compiled_base.c (hunk 1 above)
Comment (skwbc) [inline numpy/core/src/multiarray/compiled_base.c]: I modified indentation here by editing 2nd commit (git rebase -i).

Comment (seberg) [inline numpy/core/src/multiarray/compiled_base.c]: Thanks, will merge soon unless someone else wants to chime in I think. Remind me if I forget ;).

**Comment by charris** [comment]
Thanks @skwbc .

**Pull request merged**